tribits/ci_support/CDashQueryAnalyzeReport.py

```diff
@@ -565,11 +565,21 @@ def stripWhiltespaceFromStrList(strListInOut):
   ('group', 'site', 'buildname')
 
 
-def getExpectedBuildsListfromCsvFile(expectedBuildsFileName):
+def getExpectedBuildsListOfDictsfromCsvFile(expectedBuildsFileName):
   return readCsvFileIntoListOfDicts(expectedBuildsFileName,
     g_expectedBuildsCsvFileHeadersRequired)
 
 
+def getExpectedBuildsListOfDictsFromCsvFileArg(expectedBuildsFileArg):
+  expectedBuildsLOD = []
+  if expectedBuildsFileArg:
```

Comment:
Apparently the preference in the Python community would be to use
```python
if expectedBuildsFileArg is not None:
```

This was news to me when I found out about it a while ago, but having adopted the practice, I do appreciate how it eases reading the code.

Reply:
Not sure I agree with this totally with this.  The statement `if <var>` evaluates to false for the empty string is "" instead of `None` and other conditions as well (like a list with zero elements).  In fact, in this case I suspect that `expectedBuildsFileArg` is actually the empty string instead of `None` but the logic  `if <var>`  applies in either case.

So in this case, I suspect the check would be `if expectedBuildsFileArg is not ""` or the equivalent.

Reply:
Ah, right, and in that case you'd want to go with `!=` in place of `is not`, I think.